src/rust.rs

```diff
@@ -379,19 +376,20 @@ impl Runtime {
                     -> Result<(),()> {
         let filename_cstr = ffi::CString::new(filename.as_bytes()).unwrap();
         debug!("Evaluating script from {} with content {}", filename, script);
-        // SpiderMonkey does not approve of null pointers.
-        let (ptr, len) = if script.len() == 0 {
-            static empty: &'static [u8] = &[];
-            (empty.as_ptr(), 0)
-        } else {
-            (script.as_bytes().as_ptr(), script.len() as c_uint)
-        };
-        assert!(!ptr.is_null());
+
         let _ac = JSAutoRealm::new(self.cx(), glob.get());
-        let options = CompileOptionsWrapper::new(self.cx(), filename_cstr.as_ptr(), line_num);
+        let options = unsafe {
+            CompileOptionsWrapper::new(self.cx(), filename_cstr.as_ptr(), line_num)
+        };
 
         unsafe {
-            if !EvaluateUtf8(self.cx(), options.ptr, ptr as *const _, len as size_t, rval.into()) {
+            let mut source = SourceText {
+                units_: script.as_ptr() as *const _,
+                length_: script.len() as u32,
+                ownsUnits_: false,
+                _phantom_0: std::marker::PhantomData,
+            };
+            if !Evaluate2(self.cx(), options.ptr, &mut source, rval.into()) {
```

Comment:
We don't need the `Utf` marker any more?

Reply:
`source` is a  `root::JS::SourceText<root::mozilla::Utf8Unit>`, `Evaluate2` only understands UTF-8.